great_expectations/execution_engine/partition_and_sample/pandas_data_partitioner.py

```diff
@@ -225,7 +225,7 @@ def partition_on_hashed_column(
         column_name: str,
         hash_digits: int,
         batch_identifiers: dict,
-        hash_function_name: str = "md5",
+        hash_function_name: str = "sha256",
```

Comment:
It doesn't look like this is called internally anywhere and it is already parameterized. I think leaving the default as is to preserve backwards compatibility would be desired. So I would revert this change.